src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/listener/BackupJobExecutionListener.java

```diff
@@ -0,0 +1,111 @@
+    
+    private Backup backupFacade;
+    
+    private BackupExecutionAdapter backupExecution;
+
+    GeoServerConfigurationLock locker;
+
+    public BackupJobExecutionListener(Backup backupFacade, GeoServerConfigurationLock locker) {
+        this.backupFacade = backupFacade;
+        this.locker = locker;
+    }
+
+    @Override
+    public void beforeJob(JobExecution jobExecution) {
+        // Acquire GeoServer Configuration Lock in READ mode
+        locker.lock(lockType);
+        
+        this.backupExecution = backupFacade.getBackupExecutions().get(jobExecution.getId());
+    }
+
+    @SuppressWarnings("unused")
+    @Override
+    public void afterJob(JobExecution jobExecution) {
+        boolean dryRun = Boolean.parseBoolean(jobExecution.getJobParameters().getString(Backup.PARAM_DRY_RUN_MODE, "false"));
+        boolean bestEffort = Boolean.parseBoolean(jobExecution.getJobParameters().getString(Backup.PARAM_BEST_EFFORT_MODE, "false"));
+
+        try {
+            final Long executionId = jobExecution.getId();
+            
+            LOGGER.fine("Running Executions IDs : " + executionId);
+            
+            if (jobExecution.getStatus() == BatchStatus.STOPPED) {
+                backupFacade.getJobOperator().restart(executionId);
+            } else {
+                LOGGER.fine("Executions Step Summaries : " + backupFacade.getJobOperator().getStepExecutionSummaries(executionId));
+                LOGGER.fine("Executions Parameters : " + backupFacade.getJobOperator().getParameters(executionId));
+                LOGGER.fine("Executions Summary : " + backupFacade.getJobOperator().getSummary(executionId));
+
+                LOGGER.fine("Exit Status : " + backupExecution.getStatus());
+                LOGGER.fine("Exit Failures : " + backupExecution.getAllFailureExceptions());
+                
+                if (jobExecution.getStatus() == BatchStatus.COMPLETED) {
+                    JobParameters jobParameters = backupExecution.getJobParameters();
+                    Resource sourceFolder = Resources.fromURL(jobParameters.getString(Backup.PARAM_OUTPUT_FILE_PATH));
+                    BackupUtils.compressTo(sourceFolder, backupExecution.getArchiveFile());
+                }
+            }
+        // Collect errors
+        } catch (NoSuchJobException | 
+                NoSuchJobExecutionException | 
+                JobInstanceAlreadyCompleteException | 
+                JobRestartException | 
+                JobParametersInvalidException | 
+                IOException e) {
+            if(!bestEffort) {
+                this.backupExecution.addFailureExceptions(Arrays.asList(e));
+                throw new RuntimeException(e);
+            } else {
+                this.backupExecution.addWarningExceptions(Arrays.asList(e));
+            }
+        } finally {
+            // Release locks on GeoServer Configuration
+            locker.unlock(lockType);
```

Comment:
It is critical that there is no way to end up the job without unlocking the job. Have you checked that afterJob gets called even under the most unfavorable conditions? Like, say an excess of requests makes GeoServer go OOM during the backup, what then?